lib/stdlib/src/maps.erl

```diff
@@ -65,6 +67,57 @@ find(_,_) -> erlang:nif_error(undef).
 
 from_list(_) -> erlang:nif_error(undef).
 
+-spec intersect(Map1,Map2) -> Map3 when
+    Map1 :: #{Key => term()},
+    Map2 :: #{term() => Value2},
+    Map3 :: #{Key => Value2}.
+
+intersect(Map1,Map2) when is_map(Map1) andalso
+                          is_map(Map2) ->
+    intersect_with(Map1,Map2,fun intersect_combiner/3);
+intersect(Map1, Map2) ->
+    erlang:error(error_type_two_maps(Map1, Map2),
+                 [Map1,Map2]).
+
+intersect_combiner(_K,_V1,V2) ->
+    V2.
+
+-spec intersect_with(Map1,Map2,Combiner) -> Map3 when
+    Map1 :: #{Key => Value1},
+    Map2 :: #{term() => Value2},
+    Combiner :: fun((Key, Value1, Value2) -> CombineResult),
+    Map3 :: #{Key => CombineResult}.
+
+intersect_with(Map1,Map2,Combiner) when is_map(Map1) andalso
+                                        is_map(Map2) andalso
+                                        is_function(Combiner,3) ->
+    case map_size(Map1) < map_size(Map2) of
+        true ->
+            intersect_with_1(maps:keys(Map1), Map1, Map2, false, Combiner);
+        false ->
+            intersect_with_1(maps:keys(Map2), Map2, Map1, true, Combiner)
+    end;
+intersect_with(Map1,Map2,Combiner) ->
+    erlang:error(error_type_merge_intersect_with(Map1,Map2,Combiner),
+                 [Map1,Map2,Combiner]).
+
+intersect_with_1([K | Rest], Map1, Map2, MapsReversed,Combiner) ->
+    case {Map1, Map2} of
+        {#{ K := V1 }, #{ K := V2 }} ->
+            NewMap1 =
+                case MapsReversed of
+                    true ->
+                        Map1#{ K := Combiner(K, V2, V1) };
```

Comment:
I think the code would be slightly clearer if `Combiner` was wrapped in a fun that reversed the arguments, rather than flipping them here, but it's a matter of taste.

Reply:
I agree with you.

Reply:
I have refactored the code: e8c6f83